src/main/java/com/umbr3114/controllers/CommentController.java

```diff
@@ -0,0 +1,160 @@
+            return "bad";
+        }
+        commentModel = new CommentModel();
+        commentModel.setPostId(postId);
+        commentModel.setAuthorId(authorId);
+        commentModel.setBodyText(bodyText);
+        commentModel.setDate();
+        mongoCollection.save(commentModel);
+        return postId;
+    });
+
+    public static Route getComments = ((request, response) -> {
+        JacksonMongoCollection<PostModel> mongoCollectionPost =
+                new CollectionFactory<PostModel>(ServiceLocator.getService().dbService(), PostModel.class).getCollection();
+        RequestParamHelper paramHelper = new RequestParamHelper(request);
+        String postId = paramHelper.valueOf("postId");
+        String startAfterId = paramHelper.valueOf("startAfterId");
+        String date = paramHelper.valueOf("date");
+        PostModel post = mongoCollectionPost.findOne(eq("_id", new ObjectId(postId)));
+        List<CommentModel> commentModelList;
+        if(post == null){
+            return "post DNE";
+        }
+        if(startAfterId == null){
+            if(date == null){
+                Date thedate = new java.util.Date(System.currentTimeMillis());
+                DateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+                date = dateFormat.format(thedate);
+            }
+            commentModelList = getByDate(postId, date);
+        }else{
+            commentModelList = getAfterId(postId, startAfterId);
+        }
+        if(postId == null){
+            return "not ok";
+        }
+        CommentListingModel commentList = new CommentListingModel();
+        commentList.count = commentModelList.size();
+        if (commentModelList.size() > 0){
+            commentList.lastId = commentModelList.get(commentModelList.size()-1)._id.toString();
+            commentList.postId = commentModelList.get(0).getPostIdString();
+        }
+        commentList.comments = commentModelList;
+        return commentList;
+
+    });
+
+    private static List<CommentModel> getAfterId(String postId, String startAfterId){
+        JacksonMongoCollection<CommentModel> mongoCollectionComments =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        JacksonMongoCollection<PostModel> mongoCollectionPost =
+                new CollectionFactory<PostModel>(ServiceLocator.getService().dbService(), PostModel.class).getCollection();
+        FindIterable<CommentModel> iterable;
+        PostModel post = mongoCollectionPost.findOne(Filters.eq("_id", postId));
+        List<CommentModel> commentModels = new ArrayList<>();
+        if(post == null){
+            return commentModels;
+        }else{
+            iterable = mongoCollectionComments.find(Filters.and(Filters.eq("post_id", postId), Filters.gt("_id", startAfterId.toString())));
```

Comment:
this actually doesn't have to be in an else block - since IF post == null, the function is returned and the remaining lines are not executed. 

- also, we are not trying to compare postId's, we are trying to compare commentId's (since we want to page the list of comments, not the list of posts). 
- startAfterId.toString() is not required since startAfterId is already a String. also, "_id" isn't a String, but an ObjectId, so you will probably want to pass `new ObjectId(startAfterId)`.
   - this is most likely the culprit as to why `startAfterId` in `/comments/list` is broken